python/pyspark/sql/streaming/stateful_processor.py

```diff
@@ -99,25 +99,25 @@ def exists(self) -> bool:
         """
         return self._list_state_client.exists(self._state_name)
 
-    def get(self) -> Iterator[Row]:
+    def get(self) -> Iterator[Tuple]:
```

Comment:
Wait, I see we provide "Row" in ValueState.get() and we build Row instance to match with the signature. Are we going to diverge UX, or the type hint is incorrect and list state also builds Row instance per element?

We don't need to strictly match with ApplyInPandasWithState but we need to be consistent among state types in TransformWithStateInPandas.

Reply:
I'm OK either way, 1) be consistent with Tuple for state read & write 2) allow Tuple for state write but provide Row for state read to be strict with schema.

I don't even think it's a crazy idea to enforce Row for state write to be consistent. Though I kind of agree that Tuple is probably easier for users to deal with.

cc. @HyukjinKwon 
Do we have preference in PySpark for such a scenario? We internally use Row, and need to decide whether we expose Row as it is, or whatever convenient type to users and handle conversion internally.
We previously used Tuple for applyInPandasWithState so probably the preference is the latter, but wanted to double confirm before moving on.

Reply:
I'm leaning towards using Tuple to be consistent with both applyInPandasWithState and input/output types. I've also updated ValueState to return Tuple for consistency. But let me know if you have any concern on this, thanks! cc @HeartSaVioR @HyukjinKwon

Reply:
I'm OK with Tuple for user facing. We would store this as Row in state store, right? We shouldn't rely on pickling.

Reply:
Yep, we store as `Row` in state store. The pickling thing is that I followed what ApplyInPandasWithStateWriter does for groupState. It serializes `Row` state to bytes in JVM: https://github.com/apache/spark/blob/55d0233d19cc52bee91a9619057d9b6f33165a0a/sql/core/src/main/scala/org/apache/spark/sql/execution/python/ApplyInPandasWithStateWriter.scala#L182 and then deserialize using pickling in python: https://github.com/apache/spark/blob/55d0233d19cc52bee91a9619057d9b6f33165a0a/python/pyspark/sql/pandas/serializers.py#L837 Are you saying we shouldn't use `pickleSer.loads` in valueState `get()`? If so, could you share your concerns here?

An alternative I can think about is to let valueState also use arrow to transmit the state row from JVM to python side and deserialize just like list state. Let me know your thoughts on this, thanks!

Reply:
No, what I meant was that even we get Tuple from users we should convert this to Row internally before sending to JVM, and vice versa. Let's ensure this would be applied for all state types. I'm going to assume this is true.

Reply:
Yep, this is what we are doing today, will apply to all state types in the future as well, thanks for the reminder!